crates/client/rpc-core/src/utils.rs

```diff
@@ -103,3 +131,184 @@ where
+    BigInt::from_bytes_be(Sign::Plus, &value.to_bytes_be()).to_biguint().unwrap()
+}
+
+fn field_element_to_big_uint_as_hex(value: &FieldElement) -> BigUintAsHex {
+    BigUintAsHex { value: field_element_to_big_uint(value) }
+}
+
+fn entry_points_by_type_to_contract_entry_points(value: EntryPointsByType) -> ContractEntryPoints {
+    fn sierra_entry_point_to_contract_entry_point(value: SierraEntryPoint) -> ContractEntryPoint {
+        ContractEntryPoint {
+            function_idx: value.function_idx.try_into().unwrap(),
+            selector: field_element_to_big_uint(&value.selector),
+        }
+    }
+    ContractEntryPoints {
+        constructor: value.constructor.iter().map(|x| sierra_entry_point_to_contract_entry_point(x.clone())).collect(),
+        external: value.external.iter().map(|x| sierra_entry_point_to_contract_entry_point(x.clone())).collect(),
+        l1_handler: value.l1_handler.iter().map(|x| sierra_entry_point_to_contract_entry_point(x.clone())).collect(),
+    }
+}
+
+// Utils to convert Casm contract class to Compiled class
+pub fn get_casm_cotract_class_hash(casm_contract_class: &CasmContractClass) -> FieldElement {
+    let compiled_class = casm_contract_class_to_compiled_class(casm_contract_class);
+    compiled_class.class_hash().unwrap()
+}
+
+/// Converts a [CasmContractClass] to a [CompiledClass]
+pub fn casm_contract_class_to_compiled_class(casm_contract_class: &CasmContractClass) -> CompiledClass {
+    CompiledClass {
+        prime: casm_contract_class.prime.to_string(),
+        compiler_version: casm_contract_class.compiler_version.clone(),
+        bytecode: casm_contract_class.bytecode.iter().map(|x| biguint_to_field_element(&x.value)).collect(),
+        entry_points_by_type: casm_entry_points_to_compiled_entry_points(&casm_contract_class.entry_points_by_type),
+        hints: vec![],        // not needed to get class hash so ignoring this
+        pythonic_hints: None, // not needed to get class hash so ignoring this
+    }
+}
+
+/// Converts a [CasmContractEntryPoints] to a [CompiledClassEntrypointList]
+fn casm_entry_points_to_compiled_entry_points(value: &CasmContractEntryPoints) -> CompiledClassEntrypointList {
+    CompiledClassEntrypointList {
+        external: value.external.iter().map(|x| casm_entry_point_to_compiled_entry_point(&x)).collect(),
+        l1_handler: value.l1_handler.iter().map(|x| casm_entry_point_to_compiled_entry_point(&x)).collect(),
+        constructor: value.constructor.iter().map(|x| casm_entry_point_to_compiled_entry_point(&x)).collect(),
+    }
+}
+
+fn casm_entry_point_to_compiled_entry_point(value: &CasmContractEntryPoint) -> CompiledClassEntrypoint {
+    CompiledClassEntrypoint {
+        selector: biguint_to_field_element(&value.selector),
+        offset: value.offset.try_into().unwrap(),
+        builtins: value.builtins.clone(),
+    }
+}
+
+fn biguint_to_field_element(value: &BigUint) -> FieldElement {
```

Comment:
Can't we write these as `From` impl?

Reply:
`BigUint` and `FieldElement` are both foreign types so I am not able to create a `From` implementation here

Reply:
Yes the only way is to create wrapper around the types like `BigUintWrapper(BigUint)` for instance. Not sure what's the best/idiomatic approach there ? cc @LucasLvy